### PR TITLE
Add IndexVersion to MainResponse

### PR DESCRIPTION
**Pull request by thecoop**
Add `index_version` field to MainResponse

### CODE REVIEW

File: modules/rest-root/src/main/java/org/elasticsearch/rest/root/MainResponse.java [hunk 1]
```diff
@@ -39,7 +41,8 @@ public class MainResponse extends ActionResponse implements ToXContentObject {
         super(in);
         nodeName = in.readString();
         version = Version.readVersion(in);
-        transportVersion = in.getTransportVersion().onOrAfter(TransportVersion.V_8_500_014) ? TransportVersion.readVersion(in) : null;
```
Comment (thecoop) [inline modules/rest-root/src/main/java/org/elasticsearch/rest/root/MainResponse.java:42]: This should have been 19 (was already fixed in serialization)

File: modules/rest-root/src/main/java/org/elasticsearch/rest/root/MainResponse.java [hunk 2]
```diff
@@ -97,6 +109,26 @@ public void writeTo(StreamOutput out) throws IOException {
...
+        } else if (version.before(Version.V_8_10_0)) {
+            return IndexVersion.fromId(version.id).luceneVersion().toString();
+        } else {
+            return "unknown";
+        }
+    }
+
+    private String getMinimumWireCompatibilityVersion() {
+        // this is no longer computed from Version
+        return TransportVersion.current().equals(transportVersion) ? TransportVersion.MINIMUM_COMPATIBLE.toString() : "unknown";
```
Comment (rjernst) [inline modules/rest-root/src/main/java/org/elasticsearch/rest/root/MainResponse.java:124]: This seems to be an unrelated change to adding index version here. I think it should be raised separately (same for min index version). These should probably be passed along as arguments to the response, instead of looked up statically?

**Review by rjernst — APPROVED** [review]
LGTM
